aggregator/src/lib.rs

```diff
@@ -111,15 +110,23 @@ impl Aggregator {
                 .on_http(rpc_url),
         );
 
-        // Create PubSubProvider
-        let ipc_path = "/tmp/anvil.ipc";
-        let ipc = IpcConnect::new(ipc_path.to_string());
-        let pubsub_provider: Arc<RootProvider<PubSubFrontend>> = Arc::new(
-            ProviderBuilder::new()
-                .on_ipc(ipc)
+        let pubsub_provider: Arc<RootProvider<PubSubFrontend>> = match chain {
+            Chain::Anvil => {
+                let ipc = IpcConnect::new("/tmp/anvil.ipc".to_string());
```

Comment:
Shouldn't we use `ws_url` here?

Reply:
On local with anvil no it is IPC connection, for this PoC let's keep it I will check if now Alloy support WS for anvil